testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/micrometer/MicrometerPrometheusTestCase.java

```diff
@@ -0,0 +1,149 @@
+import org.junit.runner.RunWith;
+import org.wildfly.test.integration.observability.setuptask.MicrometerSetupTask;
+import org.wildfly.test.stabilitylevel.StabilityServerSetupSnapshotRestoreTasks;
+
+@RunWith(Arquillian.class)
+@ServerSetup({MicrometerSetupTask.class, StabilityServerSetupSnapshotRestoreTasks.Preview.class})
+@DockerRequired(AssumptionViolatedException.class)
+@RunAsClient
+public class MicrometerPrometheusTestCase {
+    private static final int REQUEST_COUNT = 5;
+    private final ModelNode prometheusRegistryAddress = Operations.createAddress(SUBSYSTEM, "micrometer", "registry", "prometheus");
+    private final String prometheusContext = "/prometheus";
+
+    @ArquillianResource
+    private URL url;
+
+    @ContainerResource
+    protected ManagementClient managementClient;
+
+    // The @ServerSetup(MicrometerSetupTask.class) requires Docker to be available.
+    // Otherwise the org.wildfly.extension.micrometer.registry.NoOpRegistry is installed which will result in 0 counters,
+    // and cause the test fail seemingly intermittently on machines with broken Docker setup.
+    @BeforeClass
+    public static void checkForDocker() {
+        AssumeTestGroupUtil.assumeDockerAvailable();
+    }
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(WebArchive.class, "micrometer-prometheus.war")
+                .addClasses(MicrometerApplication.class, MicrometerResource.class)
+                .addAsWebInfResource(CdiUtils.createBeansXml(), "beans.xml");
+    }
+
+    @Test
+    public void basicPrometheusTest() throws Exception {
+        addPrometheusRegistry();
+        makeRequests();
+
+        String metrics = fetchPrometheusMetrics(false);
+        Assert.assertTrue("'demo_counter_total " + REQUEST_COUNT + "' is expected",
+                metrics.contains("demo_counter_total " + REQUEST_COUNT));
+        Assert.assertTrue("'demo_timer_seconds_count' is expected",
+                metrics.contains("demo_timer_seconds_count"));
+    }
+
+    @Test
+    public void securedPrometheusTest() throws Exception {
+        setPrometheusSecurity(true);
+        makeRequests();
+
+        String metrics = fetchPrometheusMetrics(false);
+        Assert.assertTrue("'401 - Unauthorized' message is expected", metrics.contains("401 - Unauthorized"));
+
+        metrics = fetchPrometheusMetrics(true);
+        Assert.assertTrue("'demo_counter_total " + REQUEST_COUNT + "' is expected",
+                metrics.contains("demo_counter_total " + REQUEST_COUNT));
+
+        setPrometheusSecurity(false);
+        makeRequests();
+        metrics = fetchPrometheusMetrics(false);
+        Assert.assertTrue("'demo_counter_total " + REQUEST_COUNT + "' is expected",
+                metrics.contains("demo_counter_total " + REQUEST_COUNT));
+    }
+
+    private void addPrometheusRegistry() throws Exception {
+        ModelNode addOperation = Operations.createAddOperation(prometheusRegistryAddress);
```

Comment:
Rather than this method, and one of the tests calling it (since AFAIK test method order isn't guaranteed), I think I would instead override StabilityServerSetupSnapshotRestoreTasks.Preview and use that class as the setup task. Then I'd add override the `doSetup()` method and add  this code there

Reply:
Done